### PR TITLE
Fixes for issues encountered converting Java CRT

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: builder/__main__.py [hunk 1]
```diff
@@ -38,7 +37,7 @@
 ########################################################################################################################
 
 def run_action(action, env):
-
+    config = env.config
     # Set build environment from config
     env.shell.pushenv()
     for var, value in config.get('build_env', {}).items():
```
Comment (graebm) [inline builder/__main__.py]: This was crashing before, right?

It wasn't picking up some other `config` defined somewhere else? If this was the case, let's change a variable name somewhere so this mistake isn't so easy to make.

but probably this was just a crash

File: builder/actions/git.py [hunk 2]
```diff
@@ -32,15 +32,16 @@ def run(self, env):
 
         sh = env.shell
 
-        sh.exec("git", "clone", self.project.url, self.path, always=True)
+        sh.exec("git", "clone", self.project.url,
+                self.path, always=True, retries=3)
```
Comment (graebm) [inline builder/actions/git.py]: retries. fucking neat

File: builder/__main__.py (hunk 1 above)
Comment (justinboswell) [reply builder/__main__.py]: Yeah, it was picking up the one at global scope from __main__.  I got rid of that.

File: builder/actions/git.py (hunk 2 above)
Comment (justinboswell) [reply builder/actions/git.py]: I've seen every git command and apt command fail while testing. This is a necessity, I'm tired of "Well just push or run it again"

Comment (graebm) [reply builder/actions/git.py]: I was serious.
Retries ARE neat.
It was cool to see you already had support for that

**Pull request merged**